src/components/AssetsComponent.jsx

```diff
@@ -20,46 +21,66 @@ export default class AssetsComponent extends Component {
   };
 
   componentDidMount() {
-    const { activePage, assetsList } = this.props;
-
-    const assetsEmpty = isEmpty(assetsList);
-
     // TODO: fix the logic so that assets are fetched when you create an asset before fetching
     // assets, otherwise, you'll only display 1 row in assets table yet there are more than one
     // assets
-    if (assetsEmpty) {
-      this.props.getAssetsAction(activePage, this.state.limit);
-    }
+    this.loadAssets();
 
     this.props.loadAllAssetModels();
     this.props.loadDropdownAssetTypes();
   }
 
+  loadAssets = () => {
+    const { activePage, assetsList, match } = this.props;
+    const { assetStatus } = match.params;
+
+    const assetsEmpty = isEmpty(assetsList);
+    const assetStat = assetStatus === undefined ? '' : assetStatus;
+    const shouldReload = this.props.assetStatus !== assetStat;
+
+    if (shouldReload) {
+      this.props.getAssetsAction(activePage, this.state.limit, null, assetStatus);
+    }
+
+    if (assetsEmpty) {

```

Comment:
For future code, know that the `if` blocks should be combined using a Logical OR. That is:
```jsx
if (shouldReload || assetsEmpty) {
  this.props.getAssetsAction(
    activePage,
    this.state.limit,
    null,
    status
  );
}
```
However, we have moved `shouldReload` and `assetsEmpty` to the container and combined them into one using `shouldFetchAssets` variable. Therefore, we'll only have one `if` block here. I.e.
```jsx
if (shouldFetchAssets) {
  this.props.getAssetsAction(...);
}

Reply:
Yeah. I realized this after I had already pushed it and forgot to correct it. Sorry about that.